### PR TITLE
[doc] How-to: Run the Universal Discovery

### PR DESCRIPTION
**Pull request by dpobel**
Documentation for https://jira.ez.no/browse/EZP-27657

### CODE REVIEW

**Review by yannickroger — APPROVED** [review]

**Review by StephaneDiot — APPROVED** [review]

**Review by SylvainGuittard — APPROVED** [review]

File: docs/howto/run_universal_discovery.md [hunk 1]
```diff
@@ -0,0 +1,281 @@
+# How-to: Run the Universal Discovery
+
+The Hybrid Platform UI application provides a widget called Universal Discovery
+(often abbreviated <abbr title="Universal Discovery">UD</abbr> or <abbr
+title="Universal Discovery Widget">UDW</a>). It is designed to be a handy way to
+pick one or several Content items/Locations in the repository. The application
+recognizes a markup convention that allows to run and configure the Universal
```
Comment (DominikaK) [inline docs/howto/run_universal_discovery.md]: recognizes a markup convention that allows **you** to run and configure the Universal

File: docs/howto/run_universal_discovery.md [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+  Discovery.
+* `data-ud-multiple`: it's a boolean attribute. If it is added, the Universal
+  Discovery will allow several Content items/Locations to be picked, otherwise
+  only one can be selected.
+* `data-ud-starting-location-id`: if set, its value will be used as the starting
+  Location id when browsing the repository
+* `data-ud-confirm-label`: if set, its value can be used to define the Confirm
+  button label. If not specified, the button label is *Confirm*.
+* `data-ud-min-discover-depth`: it can be used to set the min discover depth of
+  the Universal Discovery. If set with a starting Location id, this allows to
```
Comment (DominikaK) [inline docs/howto/run_universal_discovery.md]: the Universal Discovery. If set with a starting Location id, this allows **you** to

File: docs/howto/run_universal_discovery.md [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+  retrict browsing to a given subtree.
```
Comment (DominikaK) [inline docs/howto/run_universal_discovery.md]: re**s**trict browsing to a given subtree.

File: docs/howto/run_universal_discovery.md [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+<button type="button" class="ez-js-run-universal-discovery"
+    data-ud-title="Pick something only under /Media/Images"
+    data-ud-starting-location-id="51"
+    data-ud-min-discover-depth="2">Run UDW</button>
+```
+
+### Restrict selection
+
+The *run Universal Discovery* convention allows the UD to be configured to only
+allow *container* Content items and/or Content items of a Content Types set to
```
Comment (DominikaK) [inline docs/howto/run_universal_discovery.md]: allow *container* Content items and/or Content items belonging to a set of Content Types to

File: docs/howto/run_universal_discovery.md [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+
+```html
+<button type="button" class="ez-js-run-universal-discovery"
+    data-ud-title="Pick a container"
+    data-ud-container>Run UDW</button>
+```
+
+#### Content items of a Content Types set
+
+The `data-ud-content-type-identifiers` can be used to let the user to only pick
```
Comment (DominikaK) [inline docs/howto/run_universal_discovery.md]: The `data-ud-content-type-identifiers` can be used to let the user only pick

File: docs/howto/run_universal_discovery.md [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+Content items of a Content Types set. This attribute expects a space separated
```
Comment (DominikaK) [inline docs/howto/run_universal_discovery.md]: Content items of a Content Type set. This attribute expects a space-separated

File: docs/howto/run_universal_discovery.md [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+#### Custom restriction
+
+To implement more advanced restrictions, the *run Universal Discovery* mechanism
+also exposes the `ez:runUniversalDiscovery:select` JavaScript event.  This event
+is dispatched from the *run Universal Discovery element* when the user is about
+to pick a Content item/Location. It is configured [to
+*bubble*](https://www.w3.org/TR/DOM-Level-2-Events/events.html#Events-flow-bubbling)
+and to [be
+*cancelable*](https://www.w3.org/TR/DOM-Level-2-Events/events.html#Events-flow-cancelation).
+Preventing the default behaviour of this event allows to restrict the user from
```
Comment (DominikaK) [inline docs/howto/run_universal_discovery.md]: Preventing the default behavior of this event allows **you** to restrict the user from

File: docs/howto/run_universal_discovery.md [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+<p id="selection-object">The event detail provides a selection object containing:</p>
+
+* the Content item object under `content` as generated by
+  [`eZ.Content#toObject`](https://github.com/ezsystems/PlatformUIBundle/blob/2.0/Resources/public/js/models/ez-contentmodel.js#L468)
+* the Location object under `location` as generated by
+  [`eZ.Location#toObject`](https://github.com/ezsystems/PlatformUIBundle/blob/2.0/Resources/public/js/models/ez-locationmodel.js#L423)
+* the Content Type object under `contentType` as generated by
+  [`eZ.ContentType#toObject`](https://github.com/ezsystems/PlatformUIBundle/blob/2.0/Resources/public/js/models/ez-contenttypemodel.js#L188)
+
+Example: only allow Content items with a Field which identifier is `image`.
```
Comment (DominikaK) [inline docs/howto/run_universal_discovery.md]: Example: only allow Content items with a Field **whose** identifier is `image`.

File: docs/howto/run_universal_discovery.md [hunk 9]
```diff
@@ -0,0 +1,281 @@
...
+
+    if ( !hasAnImageField(selection) ) {
+        e.preventDefault();
+    }
+});
+```
+
+### Use the Universal Discovery selection
+
+The *run Universal Discovery* convention allows to handle the Universal
```
Comment (DominikaK) [inline docs/howto/run_universal_discovery.md]: The *run Universal Discovery* convention allows **handling** the Universal

File: docs/howto/run_universal_discovery.md [hunk 10]
```diff
@@ -0,0 +1,281 @@
...
+Discovery selection validated by the user. As for the selection restriction, a
+built-in behavior can be configured and a JavaScript event allows to implement
```
Comment (DominikaK) [inline docs/howto/run_universal_discovery.md]: built-in behavior can be configured and a JavaScript event allows **you** to implement

**Review by DominikaK — COMMENTED** [review]
Small language nitpicks

**Review by DominikaK — APPROVED** [review]

**Pull request merged**